src/select.rs

```diff
@@ -6,13 +6,19 @@ use std::ascii::AsciiExt;
 use string_cache::{Atom, Namespace};
 use tree::{NodeRef, NodeData, ElementData};
 
+/// The definition of whitespace per CSS Selectors Level 3 § 4.
+///
+/// Copied from rust-selectors.
+static SELECTOR_WHITESPACE: &'static [char] = &[' ', '\t', '\n', '\r', '\x0C'];
+
+#[derive(Debug)]
 pub struct KuchikiSelectors;
 
 impl SelectorImpl for KuchikiSelectors {
     type NonTSPseudoClass = PseudoClass;
     type PseudoElement = PseudoElement;
```

Comment:
We could take this opportunity to simplify Kuchiki and use `String` everywhere, but string-cache makes comparison faster and selector matching makes a lot of string comparison. This was a very significant performance improvement in Servo, but I don’t know if it’s as critical in Kuchiki.